src/components/elements/MapPolygonPanel/MapPolygonCheckPanel.tsx

```diff
@@ -13,19 +16,86 @@ export interface MapPolygonCheckPanelProps {
   emptyText?: string;
   onLoadMore: () => void;
   selected: MapMenuPanelItemProps | undefined;
+  mapFunctions: any;
 }
 
-const MapPolygonCheckPanel = ({ emptyText, onLoadMore, selected }: MapPolygonCheckPanelProps) => {
+interface CheckedPolygon {
+  uuid: string;
+  valid: boolean;
+  checked: boolean;
+  nonValidCriteria: Array<object>;
+}
+
+const validationLabels: any = {
+  3: "Overlapping Polygon",
+  4: "Self-Intersection",
+  6: "Not Inside Size Limit",
+  7: "No Within Country",
+  8: "Spikes",
+  10: "Polygon Type",
+  12: "No Within Total Area Expected",
+  14: "No Data Completed"
+};
+
+const MapPolygonCheckPanel = ({ emptyText, onLoadMore, selected, mapFunctions }: MapPolygonCheckPanelProps) => {
   const t = useT();
 
   const refContainer = useRef<HTMLDivElement>(null);
+
+  const { siteData } = useMapAreaContext();
+  const context = useSitePolygonData();
+  const [polygonsValidationData, setPolygonsValidationData] = useState<any[]>([]);
+  const sitePolygonData = context?.sitePolygonData ?? [];
+
+  const { data: currentValidationSite } = useGetV2TerrafundValidationSite<CheckedPolygon[]>(
+    {
+      queryParams: {
+        uuid: siteData?.uuid ?? ""
+      }
+    },
+    {
+      enabled: !!siteData?.uuid
+    }
+  );
+
+  const parseData = (
```

Comment:
This function doesn't seem to rely on anything in the state of this component or the props; please move it out of the component.

Reply:
Done